packages/components/src/navigation/back-button/index.js

```diff
@@ -29,15 +29,23 @@ function NavigationBackButton(
 
 	const parentMenuTitle = navigationTree.getMenu( parentMenu )?.title;
 
+	const handleOnClick = ( event ) => {
```

Comment:
Does it make sense to add `useCallback` here to maintain a single reference unless a dependency changes?

Reply:
I didn't profile it, but I think `useCallback` would make this worse. This function is simple and it is only used in one component, so we would only save 1 rerender which is probably cheaper than `useCallback`. 🤔

Reply:
Sounds good!